ruby/command-t/scanner/file_scanner.rb

```diff
@@ -81,35 +102,11 @@ def ensure_cache_under_limit
         @paths_keys << @path
       end
 
-      def path_excluded?(path, prefix_len = @prefix_len)
-        if apply_wild_ignore?
```

Comment:
This is the short-circuit that I mentioned in my top-level comment, which can be used to avoid hitting the `expand()` code-path entirely.